sw/device/lib/testing/flash_ctrl_testutils.c

```diff
@@ -286,26 +287,54 @@ bool flash_ctrl_testutils_bank_erase(dif_flash_ctrl_state_t *flash_state,
   return retval;
 }
 
-uint32_t flash_ctrl_testutils_get_count(uint32_t *strike_counter) {
-  uint32_t addr = (uint32_t)strike_counter;
-  uint32_t val = abs_mmio_read32(addr);
-
-  for (size_t i = 0; i < 32; ++i) {
-    if (val >> i & 0x1) {
-      return i;
+enum {
+  kNonVolatileCounterFlashWords = 256,
+};
+static_assert(
+    FLASH_CTRL_PARAM_BYTES_PER_WORD == sizeof(uint64_t),
+    "Elements of the counter array must be the same size as a flash word");
+extern const char _non_volatile_counter_flash_words[];
+
+OT_SECTION(".non_volatile_counter_0")
+uint64_t nv_counter_0[kNonVolatileCounterFlashWords];
+OT_SECTION(".non_volatile_counter_1")
+uint64_t nv_counter_1[kNonVolatileCounterFlashWords];
+OT_SECTION(".non_volatile_counter_2")
+uint64_t nv_counter_2[kNonVolatileCounterFlashWords];
+
+static uint64_t *const kNvCounters[] = {
+    nv_counter_0,
+    nv_counter_1,
+    nv_counter_2,
+};
+
+uint32_t flash_ctrl_testutils_counter_get(size_t counter) {
+  CHECK(counter < ARRAYSIZE(kNvCounters));
+  CHECK((uint32_t)&_non_volatile_counter_flash_words ==
+        kNonVolatileCounterFlashWords);
```

Comment:
Is this what you actually meant to test?

Reply:
Correct, this verifies that the size used both in the source code and the linker script matches. Since `_non_volatile_counter_flash_words` is not a compile-time constant this has to be done at runtime.